sources/packages/api/src/services/institution-location/institution-location.service.ts

```diff
@@ -217,4 +219,35 @@ export class InstitutionLocationService extends RecordDataModelService<Instituti
       .where("location.id = :locationId", { locationId })
       .getRawOne();
   }
+
+  /**
+   * Get an institution's locations with designation statuses.
+   * @param institutionId institution id
+   * @returns Institution Locations with designation statuses.
+   */
+  async getInstitutionLocationsWithDesignationStatus(
+    institutionId: number,
+  ): Promise<LocationWithDesignationStatus[]> {
+    return this.repo
+      .createQueryBuilder("location")
+      .addSelect("location.name", "locationName")
+      .addSelect("location.info", "locationAddress")
+      .addSelect("location.id", "id")
+      .addSelect("location.institution_code", "institutionCode")
+      .addSelect("location.primary_contact", "primaryContact")
+      .innerJoin("location.institution", "institution")
+      .where("institution.id = :id", {
+        id: institutionId,
+      })
+      .addSelect(
```

Comment:
Please keep the "selects" grouped one right after the other in order to make the code easier to read.